src/lib/server/links/utils.ts

```diff
@@ -1,15 +1,70 @@
 import QRCode from 'qrcode';
-import { LINKS } from './data';
+import type { Link, ParsedLink } from './types';
 
-export function parseLinkId(link: string): keyof typeof LINKS | undefined {
-  const cleanedLinkName = link
+export function parseLink(url: string): ParsedLink | undefined {
+  const hashBeginIdx = url.lastIndexOf('#');
+  let hashStr = '';
+  if (hashBeginIdx !== -1) {
+    hashStr = url.slice(hashBeginIdx + 1);
+    url = url.slice(0, hashBeginIdx);
+  }
+
+  const queryBeginIdx = url.indexOf('?');
+  let queryStr = '';
+  if (queryBeginIdx !== -1) {
+    queryStr = queryBeginIdx === -1 ? '' : url.slice(queryBeginIdx + 1);
+    url = url.slice(0, queryBeginIdx);
+  }
+
+  const relativePathnameStr = url
     .replace(/^\/+|\/+$/g, '')
     .replace(/[^a-zA-Z0-9-/]/g, '')
     .toLowerCase();
 
-  if (!(cleanedLinkName in LINKS)) return;
+  return { relativePathname: relativePathnameStr, query: queryStr, hash: hashStr };
+}
+
+export function findLinkId<ID extends string>(
```

Comment:
document this function

Reply:
Resolved in <https://github.com/EthanThatOneKid/acmcsuf.com/pull/607/commits/b7d7b68bbfe8a9ba96f567b21907f756c5cf101f>.

Reply:
no jsdoc format?

Reply:
Nice catch; I added JSDoc formatting.